### PR TITLE
Remove declared NodePort service type in dex, remove nodeports in ist…

### PR DESCRIPTION
**Pull request by piotrmsc**
…io gateway

**Description**

Changes proposed in this pull request:

- Removed nodeports from ingress gateway
- Changed dex-service service type

**Related issue(s)**

#4852

### CODE REVIEW

File: resources/istio/charts/gateways/values.yaml [hunk 1]
```diff
@@ -61,14 +61,8 @@ istio-ingressgateway:
   - port: 80
```
Comment (piotrmsc) [inline resources/istio/charts/gateways/values.yaml]: Ports are also overriden in values.yaml but I deleted it explicitly as well

File: resources/dex/templates/upgrade-job.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  name: dex-upgrade-helper
+  annotations:
+    "helm.sh/hook-delete-policy": "before-hook-creation, hook-succeeded"
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "10"
+spec:
+  backoffLimit: 1
+  template:
+    metadata:
+      name: kyma-ns-label
```
Comment (Demonsthere) [inline resources/dex/templates/upgrade-job.yaml]: Copy&Paste error

Comment (piotrmsc) [reply resources/dex/templates/upgrade-job.yaml]: true, changed.

**Pull request merged**